src/Plugins/RpcServer/Model/ContractNameOrHashOrId.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ContractNameOrHashOrId.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Neo.Plugins.RpcServer.Model;
+
+public class ContractNameOrHashOrId
+{
+    private readonly object _value;
+
+    public ContractNameOrHashOrId(int id)
+    {
+        _value = id;
+    }
+
+    public ContractNameOrHashOrId(UInt160 hash)
+    {
+        _value = hash;
+    }
+
+    public ContractNameOrHashOrId(string name)
+    {
+        _value = name;
+    }
+
+    public bool IsId => _value is int;
+    public bool IsHash => _value is UInt160;
+    public bool IsName => _value is string;
+
+    public static bool TryParse(string value, [NotNullWhen(true)] out ContractNameOrHashOrId contractNameOrHashOrId)
+    {
+        if (int.TryParse(value, out var id))
+        {
+            contractNameOrHashOrId = new ContractNameOrHashOrId(id);
+            return true;
+        }
+        if (UInt160.TryParse(value, out var hash))
+        {
+            contractNameOrHashOrId = new ContractNameOrHashOrId(hash);
+            return true;
+        }
```

Comment:
Why not to try `ToScriptHash()` right here if `UInt160.TryParse` wasn't successful? It allows you not to make it manually in the handlers code.

Reply:
Because it's the same thing, one without throwing error.

Reply:
`UInt160.TryParse` won't parse the address.

Reply:
Logic remains the same, it can be many ways to implement and can be polished in the future. Currently in this pr there is only one rpc method need to handle scripthash/address, thus i implement it this way to keep it simple. 

Again, tool class can be updated in the following pr that adds more rpc methods.